src/plugins/ResizeReorder/ResizerKnob.js

```diff
@@ -0,0 +1,205 @@
+import FixedDataTableEventHelper from '../../FixedDataTableEventHelper';
+import ResizerLine from './ResizerLine';
+import clamp from '../../vendor_upstream/core/clamp';
+import DOMMouseMoveTracker from '../../vendor_upstream/dom/DOMMouseMoveTracker';
+import _ from 'lodash';
+
+class ResizerKnob extends React.PureComponent {
+  initialState = {
+    /**
+     * @type {boolean} Set true when column resizing starts. It is used to make ResizerLine visible.
+     */
+    isColumnResizing: false,
+
+    /**
+     * @type {number} X coordinate of ResizerLine during resizing. It is passed do to ResizerLine to render at appropriate position.
+     */
+    currentMouseXCoordinate: 0,
+
+    /**
+     * @type {number} Total displacement of mouse calculated from initial position when resizing started
+     */
+    totalDisplacement: 0,
+
+    /**
+     * @type {number} Top position of ResizerKnow. It is passed to ResizerLine to render at appropriate position.
+     */
+    top: 0,
+  };
+
+  state = { ...this.initialState };
+
+  /**
+   * Ref to ResizerKnob
+   * @type {HTMLDivElement}
+   */
+  curRef = null;
+
+  /**
+   *
+   * @type {DOMMouseMoveTracker}
+   */
+  mouseMoveTracker = null;
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.setState({
+      top: this.curRef.getBoundingClientRect().top,
+    });
+  }
+
+  render() {
+    const resizerKnobStyle = {
+      height: this.props.height,
+    };
+    const resizerLine = (
+      <ResizerLine
+        height={this.props.resizerLineHeight}
+        visible={!!this.state.isColumnResizing}
+        xCoordinate={this.state.currentMouseXCoordinate}
+        top={this.state.top}
+      />
+    );
+
+    return (
+      <div
+        className={cx('fixedDataTableCellLayout/columnResizerContainer')}
+        ref={(element) => (this.curRef = element)}
+        style={resizerKnobStyle}
+        onMouseDown={this.onMouseDown}
+        onTouchStart={this.props.touchEnabled ? this.onMouseDown : null}
+        onTouchEnd={this.props.touchEnabled ? this.suppressEvent : null}
+        onTouchMove={this.props.touchEnabled ? this.suppressEvent : null}
+      >
+        {resizerLine}
+      </div>
+    );
+  }
+
+  /**
+   * Registers event listeners for mouse tracking
+   * @param {MouseEvent} event
+   */
+  initializeDOMMouseMoveTracker = (event) => {
+    this.mouseMoveTracker = new DOMMouseMoveTracker(
+      this.onMouseMove,
+      this.onMouseUp,
+      document.body,
+      this.props.touchEnabled,
+    );
+    this.mouseMoveTracker.captureMouseMoves(event);
+  };
+
+  /**
+   * @param {MouseEvent} ev Mouse down event
+   */
+  onMouseDown = (ev) => {
+    this.initializeDOMMouseMoveTracker(ev);
+    const initialMouseXCoordinate = FixedDataTableEventHelper.getCoordinatesFromEvent(ev).x;
+    this.setState({
+      initialMouseXCoordinate,
+      isColumnResizing: true,
+      totalDisplacement: 0,
+      currentMouseXCoordinate: initialMouseXCoordinate,
```

Comment:
do we need to store both these state fields? 
Can't `totalDisplacement` be calculated through `initialMouseXCoordinate` and `currentMouseXCoordinate`?